src/qrscanner.py

```diff
@@ -15,42 +16,56 @@
 except:
     pass
 
-# OK responce from scanner
+# OK response from scanner
 SUCCESS        = b"\x02\x00\x00\x01\x00\x33\x31"
 # serial port mode
 SERIAL_ADDR    = b"\x00\x0D"
 SERIAL_VALUE   = 0xA0 # use serial port for data
-# command mode, light settings
+
+""" We switch the scanner to continuous mode to initiate scanning and
+to command mode to stop scanning. No external trigger is necessary """
 SETTINGS_ADDR  = b"\x00\x00"
-SETTINGS_VALUE = 0xD1 # use command mode + aim light, no flash light
+SETTINGS_CMD_MODE = 0xD1 # use command mode + aim light, no flash light
+SETTINGS_CONT_MODE = 0xD2 # use continuous mode + aim light, no flash light
+
+""" After basic scanner configuration (9600 bps) uart is set to 115200 bps
+to support fast scanning of animated qrs """
+BAUD_RATE_ADDR = b"\x00\x2A"
+BAUD_RATE = b"\x1A\x00" # 115200
+
 # commands
 SCAN_ADDR      = b"\x00\x02"
 # timeout
 TIMOUT_ADDR    = b"\x00\x06"
 
+""" After the scanner obtains a scan it waits 100ms and starts a new scan."""
+INTERVAL_OF_SCANNING_ADDR = b"\x00\x05"
+INTERVAL_OF_SCANNING = 0x01 # 100ms
+
+""" DELAY_OF_SAME_BARCODES of 5 seconds means scanning the same barcode again
+(and sending it over uart) can happen only when the interval times out or resets
+which happens if we scan a different qr code. """
+DELAY_OF_SAME_BARCODES_ADDR = b"\x00\x13"
+DELAY_OF_SAME_BARCODES = 0x85 # 5 seconds
+
+
 class QRScanner:
+    uart_bus = None
     def __init__(self, trigger=None, uart="YA", baudrate=9600):
         if simulator:
             self.EOL = "\r\n"
+            self.UART_EMPTY = b""
         else:
             self.EOL = "\r"
+            self.UART_EMPTY = None
         
+        QRScanner.uart_bus = uart
         self.uart = pyb.UART(uart, baudrate, read_buf_len=2048)
-        self.uart.read(self.uart.any())
 
         self.trigger = None
         self.is_configured = False
-        # we autoconfigure scanner before the first scan, 
-        # as we don't know if scanner already got power or not
-        # during the constructor
-        # and if we have scanner at all
-        if trigger is not None or simulator:
-            self.trigger = pyb.Pin(trigger, pyb.Pin.OUT)
-            self.trigger.on()
-            self.is_configured = True
-
+        self.init()
```

Comment:
i havent noticed any problems with initializing here in constructor. If you know there are problems i rather call this as a callback after 2 seconds